### PR TITLE
Add helm index to docs

### PR DESCRIPTION
**Pull request by lwr20**
## Description

Adds a helm index to the docs.  helm3 users can now install by:
```
helm repo add projectcalico https://docs.projectcalico.org/charts
helm install <version> projectcalico/calico 
```
and upgrade with:
```
helm upgrade <version> projectcalico/calico 
```

## Todos

- [x] Tests
- [x] Documentation
- [x] Release note

## Release Note

```release-note
Adds a helm index to the docs.  helm users can now install by:
`helm repo add projectcalico https://docs.projectcalico.org/charts`
`helm install <version> projectcalico/calico`
```

### CODE REVIEW

**Review by bcreane — APPROVED** [review]
Looks good, a couple of suggestions which I think would be nice to add. 
Please also squash before merging.

**Comment by lwr20** [comment]
I'm not seeing the suggestions?

File: Makefile [hunk 1]
```diff
@@ -376,6 +376,17 @@ release-publish: release-prereqs $(UPLOAD_DIR)
...
 
+## Updates helm-index with the new release chart
+helm-index: release-prereqs
+	mkdir -p charts/$(CALICO_VER)/
+	cd charts/
+	cp $(RELEASE_HELM_CHART) $(CALICO_VER)/
+	wget https://s3.amazonaws.com/calico-public/index.html -O index.yaml.bak
+	helm repo index . --merge index.yaml.bak --url https://github.com/projectcalico/calico/releases/download/
+	aws s3 cp index.html s3://calico-public/index.html
+	cd ..
```
Comment (bcreane) [inline Makefile:387]: suggest `cd -`

**Comment by bcreane** [comment]
The other suggestion appears to no longer be relevant - add a comment `<-- This file is autogenerated by helm, do no edit -->` to `index.html`. I don't see that file in the PR anymore.

**Comment by bcreane** [comment]
actually adding a comment, unless its directly supported by `helm` is probably not worth the bother since you have to check for the presence of the comment (unless `merge` always wipes it out) and only prepend when it's not there already. 

So :shipit:

File: Makefile [hunk 2]
```diff
@@ -376,6 +376,15 @@ release-publish: release-prereqs $(UPLOAD_DIR)
 	@echo "  https://github.com/projectcalico/calico/releases/tag/$(CALICO_VER)"
 	@echo ""
 
+## Updates helm-index with the new release chart
+helm-index: release-prereqs
+	mkdir -p charts/$(CALICO_VER)/
```
Comment (mgleung) [inline Makefile:381]: nit: I'm assuming if something fails (like someone didn't have the helm credentials and this was kicked off) that it's possible that the `charts/` directory will be leftover from a previous run. May want to add a `rm -rf charts` above this line just for safety.

**Review by mgleung — APPROVED** [review]
LGTM. Just one nit, but nothing that should block merging this.

File: Makefile (hunk 2 above)
Comment (lwr20) [reply Makefile:381]:
```suggestion
	rm -rf  charts
	mkdir -p charts/$(CALICO_VER)/
```

**Comment by gclawes** [comment]
Should the `helm upgrade` command above be referencing `projectcalico/calico` or `projectcalico/tigera-operator`?

**Comment by lwr20** [comment]
> Should the `helm upgrade` command above be referencing `projectcalico/calico` or `projectcalico/tigera-operator`?

You are correct, I've updated it.  I'd fixed that in the docs themselves https://docs.projectcalico.org/master/maintenance/kubernetes-upgrade#upgrading-an-installation-that-was-installed-using-helm but failed to get the comment above right.

**Pull request merged**